File: Java/Advanced Conditional Statements/Exercise/ACS Exercise/src/FishingBoat.java
import java.util.Scanner;

public class FishingBoat {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int budget = Integer.parseInt(input.nextLine());
        String season = input.nextLine();
        int fishers = Integer.parseInt(input.nextLine());

        double boatRent = 0;

        switch(season){
            case "Spring":
                boatRent = 3000;

                if(fishers <= 6){
                    boatRent -= boatRent * 0.1;
                }else if(fishers >= 7 && fishers <= 11){
                    boatRent -= boatRent * 0.15;
                }else if(fishers >= 12){
                    boatRent -= boatRent * 0.25;
                }

                if(fishers % 2 == 0){
                    boatRent -= boatRent * 0.05;
                }
                break;
            case "Summer":
                boatRent = 4200;

                if(fishers <= 6){
                    boatRent -= boatRent * 0.1;
                }else if(fishers >= 7 && fishers <= 11){
                    boatRent -= boatRent * 0.15;
                }else if(fishers >= 12){
                    boatRent -= boatRent * 0.25;
                }

                if(fishers % 2 == 0){
                    boatRent -= boatRent * 0.05;
                }
                break;
            case "Autumn":
                boatRent = 4200;

                if(fishers <= 6){
                    boatRent -= boatRent * 0.1;
                }else if(fishers >= 7 && fishers <= 11){
                    boatRent -= boatRent * 0.15;
                }else if(fishers >= 12){
                    boatRent -= boatRent * 0.25;
                }

                break;
            case "Winter":
                boatRent = 2600;

                if(fishers <= 6){
                    boatRent -= boatRent * 0.1;
                }else if(fishers >= 7 && fishers <= 11){
                    boatRent -= boatRent * 0.15;
                }else if(fishers >= 12){
                    boatRent -= boatRent * 0.25;
                }

                if(fishers % 2 == 0){
                    boatRent -= boatRent * 0.05;
                }
        }

        if(boatRent <= budget){
            System.out.printf("Yes! You have %.2f leva left.", (double)budget - boatRent);
        }else {
            System.out.printf("Not enough money! You need %.2f leva.", boatRent - budget);
        }

    }
}

File: Python/Basics/Conditional Statements Advanced/Exercise/FishingBoat.py
groupBudget = int(input())
season = input()
groupSize = int(input())

costs = 0

boatRent = [3000, 4200, 2600] #spring summer/autumn winter
sales = [10, 15, 25] #[1,6] [7,11] [12># ]

evenSizeSale = 5 #pct

if season == "Spring":
    costs += boatRent[0]
elif season == "Summer" or season == "Autumn":
    costs += boatRent[1]
elif season == "Winter":
    costs += boatRent[2]


if 0 < groupSize <= 6:
    costs -= costs * (sales[0] / 100)
elif 7 <= groupSize <= 11:
    costs -= costs * (sales[1] / 100)
elif 12 <= groupSize:
    costs -= costs * (sales[2] / 100)

if season == "Spring" or season == "Winter" or season == "Summer":
    if groupSize % 2 == 0:
        costs -= costs * (evenSizeSale / 100)

if costs <= groupBudget:
    print(f"Yes! You have {groupBudget - costs:.2f} leva left.")
else:
    print(f"Not enough money! You need {costs - groupBudget:.2f} leva.")